### ISSUE TITLE
1.7+ - Duplicate entries for local project dependencies

### ISSUE THREAD

**Issue by mikethecalamity** [CycloneDX/cyclonedx-gradle-plugin#261]
### Description

When using versions `1.7.x` with projects that reference other projects within the same repository, the `cyclonedxBom` task fails with a `The BOM does not conform to the CycloneDX BOM standard` error. This is due to a duplicate entry for the local project dependency in the `bom.xml` (see stacktrace below). The duplicate entries are also in `1.6.x`, but there is no `bom-ref` attribute to trigger the invalid schema.

### Behavior
The command might need to be run twice after pulling the repository.
```
$ ./gradlew app-module:build app-module:cyclonedxBom --stacktrace

> Task :app-module:cyclonedxBom FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app-module:cyclonedxBom'.
> The BOM does not conform to the CycloneDX BOM standard

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app-module:cyclonedxBom'.
        ...
Caused by: org.gradle.api.GradleException: The BOM does not conform to the CycloneDX BOM standard
        at org.cyclonedx.gradle.CycloneDxTask.writeXMLBom(CycloneDxTask.java:601)
        at org.cyclonedx.gradle.CycloneDxTask.writeBom(CycloneDxTask.java:574)
        at org.cyclonedx.gradle.CycloneDxTask.createBom(CycloneDxTask.java:317)
        ...
Caused by: org.cyclonedx.exception.ParseException: cvc-identity-constraint.4.1: Duplicate unique value [pkg:maven/org.dummy.utils/core-module@0.0.1-SNAPSHOT?type=jar] declared for identity constraint "bom-ref" of element "bom".
        at org.cyclonedx.parsers.XmlParser$1.error(XmlParser.java:214)
        ...
        at org.cyclonedx.parsers.XmlParser.validate(XmlParser.java:217)
        at org.cyclonedx.parsers.XmlParser.validate(XmlParser.java:148)
        at org.cyclonedx.gradle.CycloneDxTask.writeXMLBom(CycloneDxTask.java:595)
        ... 146 more
Caused by: org.xml.sax.SAXParseException; systemId: file:app-module/build/reports/bom.xml; lineNumber: 196; columnNumber: 103; cvc-identity-constraint.4.1: Duplicate unique value [pkg:maven/org.dummy.utils/core-module@0.0.1-SNAPSHOT?type=jar] declared for identity constraint "bom-ref" of element "bom".
        at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
        at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.error(ErrorHandlerWrapper.java:135)
        ... 170 more
```

### Reproduce
1. Pull repository https://github.com/mikethecalamity/gradle-test
2. Run `./gradlew app-module:cyclonedxBom --stacktrace`

**Comment by mikethecalamity** [comment]
@glefloch

**Comment by mikethecalamity** [comment]
@glefloch any ideas?

**Comment by mikethecalamity** [comment]
It still seems to be an issue in `1.7.4`

**Comment by mikethecalamity** [comment]
@glefloch bump?

**Comment by laudenbc** [comment]
Is there any news here? The issue still occurs with version 1.8.1. The plugin is currently not usable for multi-project builds.

**Comment by glefloch** [comment]
Sorry, I haven't been able to find time to spent on this. Thanks for the reproducer, I will integrate it tests and push a fix

**Comment by tobias-kleinschmidt-fnt** [comment]
This problem also occurs when adding the Gradle text fixtures plugin to the project: https://docs.gradle.org/7.6.2/userguide/java_testing.html#sec:java_test_fixtures

**Comment by laudenbc** [comment]
Is there any news here? All users with multi-project builds are stuck on version 1.6 and cannot update because of this bug.

**Comment by jensborrmann** [comment]
I did a quick retests with @mikethecalamity 's test repo using 1.8.2. The stacktrace no longer occurs (as opposed to 1.8.1).

What is left is a warning:
`Unable to resolve POM for org.dummy.utils:utils-module:0.0.1-SNAPSHOT: org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration$ArtifactResolveException: Could not resolve all files for configuration ':app-module:detachedConfiguration6'.
`
Unfortunately, my detailed knowledge of the CycloneDX format is not sufficient to verify the content of the result. Perhaps one of the experts can take a look at this and possibly close this bug. After all, some things happen by chance...

**Comment by skhokhlov** [comment]
I believe the reason for this was that the core library didn't implement `equals` and `hashCode` https://github.com/CycloneDX/cyclonedx-core-java/issues/409#issuecomment-2152255554. The fix was released with plugin version `1.9.0`. Please try with this version. 

Feel free to reopen this issue if it doesn't work for you.
